proto/proto-backplane-grpc/src/main/proto/deephaven/proto/config.proto

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2016-2022 Deephaven Data Labs and Patent Pending
+ */
+syntax = "proto3";
+
+package io.deephaven.proto.backplane.grpc;
+
+option java_multiple_files = true;
+option optimize_for = SPEED;
+option go_package = "github.com/deephaven/deephaven-core/go/internal/proto/config";
+
+/**
+ * Provides simple configuration data to users. Unauthenticated users may call GetAuthenticationConstants
+ * to discover hints on how they should proceed with providing their identity, while already-authenticated
+ * clients may call GetConfigurationConstants for details on using the platform.
+ */
+service ConfigService {
+  rpc GetAuthenticationConstants(AuthenticationConstantsRequest) returns (AuthenticationConstantsResponse) {}
+  rpc GetConfigurationConstants(ConfigurationConstantsRequest) returns (ConfigurationConstantsResponse) {}
+}
+
+message AuthenticationConstantsRequest {}
+message ConfigurationConstantsRequest {}
+message AuthenticationConstantsResponse {
+  repeated ConfigPair config_values = 1;
+}
+message ConfigurationConstantsResponse {
+  repeated ConfigPair config_values = 1;
+}
+message ConfigPair {
+  string key = 1;
+  oneof value {//leave room for more types
+    string string_value = 2;
+  }
+}
```

Comment:
So, I do want to talk about pros and cons of our choices here wrt https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/struct.proto. Is there benefit to relying on Struct/Value instead? Or, we could potentially mirror it with our own type and leave out the parts we don't need right now:

```proto
message Value {
  // The kind of value.
  oneof kind {
    // Represents a string value.
    string string_value = 3;
  }
}
```
This might leave us room in the future if we _did_ want to transition to upstream types.
Thoughts on `repeated` vs `map<string, `? We get a bit stronger of typing w/ a map.

Reply:
Agreed, in theory, will make the change.

I'm not sure what you mean by stronger typing though, the key is still a string the value is still whatever fits in the one-of/etc, which is still checked by tag. Indeed, the `map<k,v>` syntax is just sugar over a message with a `1` and `2` tag of the specified type, so key collisions can happen (though presumably the receiving side's runtime will prune the duplicates, like it would for non-repeated fields). Note especially the warning in the docs at https://developers.google.com/protocol-buffers/docs/proto3#backwards_compatibility:

> Any protocol buffers implementation that supports maps must both produce and accept data that can be accepted by the above definition.

Functionally, the only difference here is that the oneof is moved out to its own message type, so we just pay for an extra tag "go decode a value now" "here's the type of the data" instead of just "here's the type of the data".

Reply:
Ah, great link. I didn't realize the semantics of map were equivalent to repeated.

Reply:
Updated, PTAL.